### PR TITLE
fix: simplified error messages for invalid inputs at profile settings

### LINKED ISSUE #2464: Bug : Simplify error messages for user at Talawa User Portal
**Describe the bug**
In Talawa User portal when the user tries to update information with empty or invalid data, error in **not** user friendly way appears. Makes it hard to understand the user what required fields they are missing or giving an invalid data into. 

**To Reproduce**

1. Go to `Talawa User Portal`
2. Go to `settings` 
3. Submit data in invalid or empty format
4. Click on save changes

**Expected behavior**
Error should come one by one in a user friendly format. 

**Actual behavior**
```
Variable "$phoneNumber" got invalid value ""; Value is not a valid phone number of the form +17895551234 (7-15 digits): Variable "$grade" got invalid value ""; Value "" does not exist in "EducationGrade" enum. Variable "$empStatus" got invalid value ""; Value "" does not exist in "EmploymentStatus" enum. Variable "$maritalStatus" got invalid value ""; Value "" does not exist in "MaritalStatus" enum.
```

**Screenshots**
image

**Additional details**

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by bandhan-majumder**
**What kind of change does this PR introduce?**

A bugfix.

**Issue Number:**

Fixes #2464 

**Snapshots/Videos:**

https://www.loom.com/share/5e08e9042d43429788277eb8b30749ea?sid=c2f89415-db97-47e9-be29-0b99386cb2de

**Passing all tests**
image
**Summary**

While updating user profile at Talawa User Portal, user used not to get user friendly errors. I have simplified the error which will come one by one in a user friendly readable manner helping user to understand the fields they are missing/giving wrong data into. I have also added translation of errors.

**Other information**
 
Added a 400 handler too.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Added multiple new error messages for input validation across English, French, Hindi, Spanish, and Chinese locales.
	- Enhanced user feedback for various input scenarios, including phone number, education grade, employment status, and marital status.

- **Bug Fixes**
	- Improved error handling logic to provide more specific feedback based on different error scenarios, including handling for status code 400.

- **Tests**
	- Introduced new test cases for the error handling function to ensure accurate feedback for various error messages.

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if any of these conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold
3) The code does not comply with our linting check or other related formatting requirements 

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by bandhan-majumder** [comment]
This should be passing now. This is coming because one function in test.tsx is accepting two parameters, but using only one. Added a console.log as a use case cause the current code is able to accept one parameter only.

**Comment by bandhan-majumder** [comment]
My code changes are totally covered. But not getting where the coverage is decreasing.Please let me know if it needs to be updated anywhere.

image

**Comment by bandhan-majumder** [comment]
HI @palisadoes. I understand that the reviewers may have been busy. I am waiting this PR to get merged so that I can start solving another issue after getting assigned.

**Comment by bandhan-majumder** [comment]
HI @palisadoes  . I understand that the reviewers may have been busy. I am waiting this PR to get merged so that I can start solving another issue after getting assigned.

**Comment by palisadoes** [comment]
Closing as abandoned.

**Comment by bandhan-majumder** [comment]
HI @palisadoes . I understand the reason of closing the PR. I unassigned myself from the issue as nobody was responding. Is there any chance of reopening it?

**Comment by palisadoes** [comment]
1. Please make sure coderabbit.ai approves your PR
1. Make sure all tests pass (except introspection)
1. Please make sure that the code coverage for your patch reaches at least 95%

File: src/utils/errorHandler.test.tsx [hunk 1]
```diff
@@ -11,23 +11,88 @@ jest.mock('react-toastify', () => ({
...
-    key;
+  const tErrors: TFunction = (
+    key: string,
+    options?: Record<string, unknown>,
+  ) => {
+    if (options) {
+      console.log(`options are passed, but the function returns only ${key}`);
+    }
+    return key;
+  };
```
[the remark this answers is not in the document]
Comment (bandhan-majumder) [reply src/utils/errorHandler.test.tsx:22]: If i remove the console.log statement, it will give error for keeping the "options" argument unused. It is better to use is as log when it could be sent with parameters, but not used in whole codebase.

**Comment by bandhan-majumder** [comment]
Thanks for reopening @palisadoes . I have increased the coverage with some edge cases. Can u please review it once?

**Comment by Cioppolo14** [comment]
@bandhan-majumder Please fix the failing tests.

**Comment by bandhan-majumder** [comment]
HI @Cioppolo14 and @palisadoes . The failing tests are graphql related and `codecov/project`. The code coverage of may changes is already passing the test `codecov/patch`. Please suggest me how to improve if those cases are necessary to pass.

File: src/utils/errorHandler.tsx [hunk 2]
```diff
@@ -5,18 +5,26 @@ import i18n from './i18n';
...
   if (error instanceof Error) {
-    switch (error.message) {
-      case 'Failed to fetch':
-        toast.error(tErrors('talawaApiUnavailable') as string);
-        break;
-      // Add more cases as needed
-      default:
-        toast.error(error.message);
+    const errorMessage = error.message;
+    if (errorMessage === 'Failed to fetch') {
```
[the remark this answers is not in the document]
Comment (bandhan-majumder) [reply src/utils/errorHandler.tsx:14]: Initially I did. But **coderabbit** suggested to make it `if-else` for better readability. Should I move it to switch case again? And one thing, after I change that, the workflow may fail the `jest` tests as it is unable to parse CSS in multiple places. Even if I just clone and run tests without any changes, it's failing. Most of the PRs are facing this issue. I also have asked @palisadoes in [slack](https://palisadoes-foundation.slack.com/archives/CSWGQ70UQ/p1734237276733659) about it.
